### ISSUE TITLE
Data type in debugger watch list hides variable value

### ISSUE THREAD

**Issue by ardnew** [golang/vscode-go#2569]
Issue Type: <b>Bug</b>

When halting any Go program under debugger `dlv`, the `Variables` and `Watch` panes in the `Run and Debug` panel show variables using their full, absolute type preceding its current value. With many nested packages, it makes values hard to see without expanding each individually.

Extension version: 0.36.0
VS Code version: Code 1.67.1 (Universal) (da15b6fd3ef856477bf6f4fb29ba1b7af717770d, 2022-05-06T12:37:16.526Z)
OS version: Darwin x64 21.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2800)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.77GB free)|
|Process Argv|. --crash-reporter-id d1db2c30-84ae-4a50-be0a-79d992e054e5|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by findleyr** [comment]
CC @suzmue @hyangah -- is there any way to improve this UX? Can we use a package-local qualifier, rather than fully-qualified symbols?

**Comment by hspaay** [comment]
+1 this is one of those issues that make debugging golang in vscode a less than optimal experience.

**Comment by firelizzard18** [comment]
I think this is a delve/dlv issue? I don't think vscode-go has control over how these are presented.

**Comment by hspaay** [comment]
> I think this is a delve/dlv issue? I don't think vscode-go has control over how these are presented.

Yeah the output is from dlv, but why can't vscode reformat that output as OP suggests? vscode manages how its presented.

**Comment by firelizzard18** [comment]
> vscode manages how its presented.

I assume you mean vscode-go, the Go extension? **No, it doesn't.** vscode-go simply connects delve to VSCode:

https://github.com/golang/vscode-go/blob/a8709218a185c201096716689b9df499c2c3be39/extension/src/goDebugFactory.ts#L58-L70

When run in "connect to an already-running instance of delve", vscode-go literally just instantiates VSCode's built in debug adapter server. When run in local/normal mode, vscode-go uses a wrapper, but that wrapper exists for one purpose only: managing the lifetime of the delve process.

So regardless of how delve is started, **delve and VSCode talk directly to each other** and presentation of delve's responses is handled by VSCode with zero involvement from vscode-go. For vscode-go to control the presentation, it would have to act as a middleman in the debug adapter protocol, parsing the text responses returned by delve and rewriting them on the fly. Doable, but it would be fragile and maintenance hell. Changing how delve presents values is a much more viable path.

**Comment by hspaay** [comment]
Oh wow, I see what you mean. 

The middleman approach is fragile because if delve changes its debug output it is going to break. That plus maintaining differences between versions. 

Thank you kindly for explaining this.
